rust_snuba/src/processors/profile_chunks.rs

```diff
@@ -44,17 +37,38 @@ struct InputMessage {
     retention_days: Option<u16>,
 }
 
-#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+#[derive(Debug, Default, Serialize, JsonSchema)]
 struct Chunk {
-    #[serde(flatten)]
-    chunk: InputMessage,
-
+    project_id: u64,
+    profiler_id: Uuid,
+    chunk_id: Uuid,
+    #[serde(rename(serialize = "start_timestamp"))]
+    start_timestamp_micro: u64,
+    #[serde(rename(serialize = "end_timestamp"))]
+    end_timestamp_micro: u64,
+    retention_days: Option<u16>,
     #[serde(default)]
     offset: u64,
     #[serde(default)]
     partition: u16,
 }
 
+impl TryFrom<FromChunkMessage> for Chunk {
+    type Error = anyhow::Error;
+
+    fn try_from(from: FromChunkMessage) -> anyhow::Result<Chunk> {
+        Ok(Self {
+            chunk_id: from.chunk_id,
+            end_timestamp_micro: (from.end_timestamp * 1e6) as u64,
+            profiler_id: from.profiler_id,
+            project_id: from.project_id,
+            retention_days: from.retention_days,
+            start_timestamp_micro: (from.start_timestamp * 1e6) as u64,
+            ..Default::default()
+        })
+    }
+}
+
```

Comment:
I concur with @Swatinem above: If you really want to implement a trait here, `From` would be the appropriate choice because the conversion is always defined.